internal/controller/controllers/agent_controller.go

```diff
@@ -282,6 +322,18 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, agent *aiv1beta1.A
 		}
 	}
 
+	err = r.updateInstallerArgs(ctx, c, internalHost, agent)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			Requeue = true
```

Comment:
Why re-queue if the host wasn't found?
Consider adding a comment why you do it...

Reply:
just took it from the example above. As i understand Requeue is needed to an option that node doesn't exists yet?
@rollandf ?